src/clj/puppetlabs/services/jruby/jruby_metrics_core.clj

```diff
@@ -277,18 +276,17 @@
 
 (schema/defn track-return-instance!
   [{:keys [return-count borrowed-instances borrow-timer] :as metrics} :- JRubyMetrics
-   {:keys [instance reason]} :- jruby-schemas/JRubyReturnedEvent]
+   {:keys [instance worker-id]} :- jruby-schemas/JRubyReturnedEvent]
   (.inc return-count)
   (when (jruby-schemas/jruby-instance? instance)
-    (let [id (:id instance)]
-      (if-let [ta (get @borrowed-instances id)]
-        (let [elapsed-time (- (System/currentTimeMillis) (:time ta))
-              per-reason-timer (timer-for-borrow-reason metrics ta)]
-          (.update borrow-timer elapsed-time (TimeUnit/MILLISECONDS))
-          (.update per-reason-timer elapsed-time (TimeUnit/MILLISECONDS))
-          (MDC/put "jruby.borrow-time" (Long/toString elapsed-time))
-          (swap! borrowed-instances dissoc id))
-        (log/warn (trs "JRuby instance ''{0}'' returned, but no record of when it was borrowed!" id))))))
+    (if-let [ta (get @borrowed-instances worker-id)]
```

Comment:
Do you have any idea what `ta` is supposed to stand for here? Is there a better name we could use?

Reply:
honestly no idea..